app/vmselect/promql/transform_test.go

```diff
@@ -220,3 +220,33 @@ func timeseriesToPromMetrics(tss []*timeseries) string {
 	}
 	return strings.Join(a, "\n")
 }
+
+func TestAlphanumericLess(t *testing.T) {
+	f := func(name, str, nextStr string, want bool) {
+		t.Helper()
+		t.Run(name, func(t *testing.T) {
+			if got := alphanumericLess(str, nextStr); got != want {
+				t.Errorf("alphanumericLess() = %v, want %v", got, want)
+			}
+		})
+	}
+	f("empty strings", "", "", false)
+	f("check that a bigger than b", "a", "b", true)
+	f("check that b lower than a", "b", "a", false)
+	f("numbers with special chars", "1:0:0", "1:0:2", true)
+	f("numbers with special chars and different number rank", "1:0:15", "1:0:2", true)
```

Comment:
This test should return `false`, since `15` is bigger than `2`.

Reply:
updated tests with new approach of the alphanumerc Function